### PR TITLE
PIA-1058: App navigation setup

### PR DESCRIPTION
**Pull request by kp-laura-sempere**
- Created `RootContainerView` that is responsible to show the correct flow based on the authentication state:
1. Splash
2. Not authenticated flow (for now only login screen)
3. Authenticated flow when Onboarding is not finished (Vpn Onboarding flow)
4. Authenticated flow when Onboarding is done (Dashboard view)

- Initial setup of `AppRouter` that will be used inside each main flow (Not authenticated and Authenticated flows) inside a `NavigationStack`:
https://developer.apple.com/documentation/swiftui/navigationstack
https://developer.apple.com/documentation/swiftui/migrating-to-new-navigation-types

NOTE: This PR depends on this other PR from PIALibrary:
https://github.com/pia-foss/mobile-ios-library/pull/24

### CODE REVIEW

File: PIA VPN-tvOS/RootContainer/Presentation/RootContainerViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        case activatedNotOnboarded
+        case activated
+    }
+    
+    @Published var state: State = .splash
+    
+    // TODO: Update this value from the Vpn OnBoarding installation profile screen
+    @AppStorage(.kOnboardingVpnProfileInstalled) var onBoardingVpnProfileInstalled = true
+    
+    let accountProvider: AccountProvider
```
Comment (kp-said-rehouni) [inline PIA VPN-tvOS/RootContainer/Presentation/RootContainerViewModel.swift:21]: I'd suggest the same as the comment above. To have something that hides all the `AccountProvider` methods and it exposes only what the ViewModel needs.

**Review by kp-said-rehouni — COMMENTED** [review]
Great job @kp-laura-sempere! Just a couple of comments to see what you think.
Thank you!

File: PIA VPN-tvOS/RootContainer/Presentation/RootContainerViewModel.swift (hunk 1 above)
Comment (kp-laura-sempere) [reply PIA VPN-tvOS/RootContainer/Presentation/RootContainerViewModel.swift:21]: Yes, I was thinking the same. I think we can create a `AccountProviderType` protocol that only contains the methods that the presentation layer needs.
Let me try to add it 👍

**Review by kp-said-rehouni — APPROVED** [review]
LGTM! Great job @kp-laura-sempere!

**Pull request merged**